test/testers/rust-tester/src/pdre_api/crypto.rs

```diff
@@ -0,0 +1,170 @@
+use super::ParsedInput;
+use super::utils::CryptoApi;
+
+use substrate_primitives::hashing::{twox_128, twox_256, twox_64};
+
+// Input: data
+pub fn test_blake2_128(input: ParsedInput) {
+    let mut api = CryptoApi::new();
+    let data = input.get(0);
+
+    let mut output = [0; 16];
+    api.rtm_ext_blake2_128(data, &mut output);
+
+    println!("Result: {}", hex::encode(output));
+}
+
+// Input: data
+pub fn test_blake2_256(input: ParsedInput) {
+    let mut api = CryptoApi::new();
+    let data = input.get(0);
+
+    let mut output = [0; 32];
+    api.rtm_ext_blake2_256(data, &mut output);
+
+    println!("Result: {}", hex::encode(output));
+}
+
+// Input: data
+pub fn test_twox_64(input: ParsedInput) {
+    let mut api = CryptoApi::new();
+    let data = input.get(0);
+
+    let mut output = [0; 8];
+
+    api.rtm_ext_twox_64(data, &mut output);
+    assert_eq!(twox_64(data), output);
+
+    println!("Result: {}", hex::encode(output));
+}
+
+// Input: data
+pub fn test_twox_128(input: ParsedInput) {
+    let mut api = CryptoApi::new();
+    let data = input.get(0);
+
+    let mut output = [0; 16];
+
+    api.rtm_ext_twox_128(data, &mut output);
+    assert_eq!(twox_128(data), output);
+
+    println!("Result: {}", hex::encode(output));
+}
+
+// Input: data
+pub fn test_twox_256(input: ParsedInput) {
+    let mut api = CryptoApi::new();
+    let data = input.get(0);
+
+    let mut output = [0; 32];
+
+    api.rtm_ext_twox_256(data, &mut output);
+    assert_eq!(twox_256(data), output);
+
+    println!("Result: {}", hex::encode(output));
+}
+
+// Input: data
+pub fn test_keccak_256(input: ParsedInput) {
+    let mut api = CryptoApi::new();
+    let data = input.get(0);
+
+    let mut output = [0; 32];
+
+    api.rtm_ext_keccak_256(data, &mut output);
+
+    println!("Result: {}", hex::encode(output));
+}
+
+// Input: data
+pub fn test_ed25519(input: ParsedInput) {
```

Comment:
While implementing this I ran into a problem; the entire process from generating, signing and verifying must be done in one function, since the storage is not persistent (the RE does not support importing keys). Honestly I'd leave it as it is and just print the values. Those keys and signatures are not deterministic anyway.

Reply:
Whatever you need to store print it out (the key, the signature),to stdout then julia can read it and feed it to the next function as a command line argument